### ISSUE TITLE
False positive "Cannot assign to immutable binding" with &mut self references

### ISSUE THREAD

**Issue by skaufhold** [intellij-rust/intellij-rust#3446]
## Environment

* **Intellij-Rust plugin version:** 0.2.0.2534-182-nightly
* **Rust toolchain version:** 1.34.0-nightly (00aae71f5 2019-02-25)
* **IDE name and version:** CLion 2018.2.6
* **Operating system:** Windows 10

## Problem description

Assigning something to an element in an indexable field of a &mut self reference yields a "Cannot assign to field of immutable binding" error when it shouldn't.

## Steps to reproduce

```rust
fn main() {
    let mut test = Test {
        stuff: vec![Test2{ a: 1}]
    };
    test.test_mut_fn();
    dbg!(test);
}

#[derive(Debug)]
struct Test {
    stuff: Vec<Test2>
}

#[derive(Debug)]
struct Test2 {
    a: isize
}

impl Test {
    pub fn test_mut_fn(&mut self) {
        let element_ref = &mut self.stuff[0];
        element_ref.a = 2; // error here: Cannot assign to field of immutable binding
    }
}
```

**Comment by ortem** [comment]
@skaufhold try to update the plugin via `Preferences -> Plugins -> Updates`

**Comment by skaufhold** [comment]
Even after checking for plugin updates explicitly, this is the most current version that I get from the Jetbrains repository.

**Comment by ortem** [comment]
Oh, you should update CLion too, because we don't support very old CLion versions.

**Comment by skaufhold** [comment]
Ah missed that this PC had a somewhat older CLion version, after updating it I get the new Plugin version which doesn't exhibit the issue. Thanks!
